dialects/core/core.thorin

```diff
@@ -76,31 +76,32 @@
 ///
 .ax %core.bit2( f,      nor,    nciff,  na,     niff,   nb,   xor_, nand,
                 and_,  nxor,    b,      iff,    a,      ciff, or_,  t):
-    Π w: .Nat -> [.Idx w, .Idx w] -> .Idx w , normalize_bit2;
+    Π s: .Nat -> [.Idx s, .Idx s] -> .Idx s , normalize_bit2;
 ///
 /// ### %core.shr
 ///
-/// Shift right - integer operation that neither takes a WMode nor produces a side effect:
-/// * [`a`rithmetic shift right](https://en.wikipedia.org/wiki/Arithmetic_shift)
-/// * [`l`ogical shift right](https://en.wikipedia.org/wiki/Logical_shift)
+/// Shift right:
+///     * [`a`rithmetic shift right](https://en.wikipedia.org/wiki/Arithmetic_shift)
+///     * [`l`ogical shift right](https://en.wikipedia.org/wiki/Logical_shift)
 ///
-.ax %core.shr(a, l): Π w: .Nat -> [.Idx w, .Idx w] -> .Idx w, normalize_shr;
+.ax %core.shr(a, l): Π s: .Nat -> [.Idx s, .Idx s] -> .Idx s, normalize_shr;
 ///
 /// ### %core.wrap
 ///
-/// Integer operations that might wrap and, hence, take a WMode.
+/// Integer operations that might wrap and, hence, take a core::Mode.
 ///
 .ax %core.wrap(add, sub, mul, shl):
-    Π [m: .Nat, w: .Nat] -> [.Idx w, .Idx w] -> .Idx w, normalize_wrap;
+    Π s: .Nat -> .Nat -> [.Idx s, .Idx s] -> .Idx s, normalize_wrap;
```

Comment:
Now that the mode no longer is bound to an identifier, the comment above the axiom should probably be more informative about the signature (what is s, what is this floating nat).